### PR TITLE
Remove -language:Scala2

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -75,10 +75,10 @@ lazy val sharedSettings = Seq(
...
 
   scalacOptions ++= Seq(
-    "-unchecked", "-deprecation", "-feature", "-Xlint",
-    "-Ywarn-adapted-args", "-Ywarn-dead-code", "-Ywarn-inaccessible",
-    "-Ywarn-nullary-override", "-Ywarn-nullary-unit",
-    "-Xlog-free-terms"
+    "-unchecked", "-deprecation", "-feature", // "-Xlint",
+    // "-Ywarn-adapted-args", "-Ywarn-dead-code", "-Ywarn-inaccessible",
+    // "-Ywarn-nullary-override", "-Ywarn-nullary-unit",
+    // "-Xlog-free-terms"
```
Comment (allanrenucci) [inline build.sbt]: I would keep these options and override the options in dottySettings. I am trying to minimise the diff with the the upstream repository to keep rebasing as simple as possible

File: shared/src/main/scala/minitest/runner/Framework.scala [hunk 2]
```diff
@@ -18,7 +18,7 @@
 package minitest.runner
 
 import minitest.runner.Framework.ModuleFingerprint
-import sbt.testing.{Framework => BaseFramework, _}
+import sbt.testing.{Framework => BaseFramework, Runner => BaseRunner, _}
```
Comment (allanrenucci) [inline shared/src/main/scala/minitest/runner/Framework.scala]: Why?

Comment (nicolasstucki) [reply shared/src/main/scala/minitest/runner/Framework.scala]: Name resolution changed, this is to hide the `sbt.testing.Running` and select the correct one bellow. In scala2 mode there is a helpful message that warns about the change.

File: build.sbt (hunk 1 above)
Comment (nicolasstucki) [reply build.sbt]: will remove the comments

File: shared/src/main/scala/minitest/runner/Framework.scala (hunk 2 above)
Comment (allanrenucci) [reply shared/src/main/scala/minitest/runner/Framework.scala]: You can do:
```scala
import sbt.testing.{Framework => BaseFramework, Runner => _, _}
```

Comment (nicolasstucki) [reply shared/src/main/scala/minitest/runner/Framework.scala]: Ok

File: build.sbt [hunk 3]
```diff
@@ -176,8 +176,7 @@ lazy val lawsJS     = laws.js
 lazy val dottyVersion = dottyLatestNightlyBuild.get
 lazy val dottySettings = List(
   scalaVersion := dottyVersion,
-  libraryDependencies := libraryDependencies.value.map(_.withDottyCompat(scalaVersion.value)),
-  scalacOptions := List("-language:Scala2")
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply build.sbt]: Done

**Pull request merged**